coops-ui/src/views/Dashboard.vue

```diff
@@ -93,7 +104,82 @@ export default {
     goToStandaloneAddresses () {
       this.setCurrentFilingStatus('NEW')
       this.$router.push({ name: 'standalone-addresses', params: { id: 0 } }) // 0 means "new COA filing"
+    },
+
+    checkToReloadDashboard () {
+      // cancel any existing timer so we can start fresh here
+      clearTimeout(this.refreshTimer)
+
+      let filingId = null
+      if (this.$route !== undefined) filingId = +this.$route.query.filing_id // if missing, this is NaN
+
+      // only consider refreshing the dashboard if we came from a filing
+      if (!filingId) return
+
+      let isInFilingHistory = this.historyFilings.filter(el => el.filingId === filingId).length > 0
+      let isInTodoList = this.todoListFilings.filter(el => el.id === filingId).length > 0
+
+      // if this filing is NOT in the to-do list and IS in the filing history list, do nothing - there is no problem
+      if (!isInTodoList && isInFilingHistory) return
+
+      // if this filing is in the to-do list, mark it as in-progress for to-do list to format differently
+      if (isInTodoList) {
+        this.inProcessFiling = filingId
+      }
+
+      // check for updated status to reload dashboard
+      this.checkFilingStatusCount = 0
+      this.checkFilingStatus(filingId)
+    },
+
+    checkFilingStatus (filingId) {
+      // check whether this filing's status has changed - recursive, runs approx. every 1 second for up to 10 seconds
+
+      this.checkFilingStatusCount++
+
+      // stop this cycle after 10 iterations
+      if (this.checkFilingStatusCount > 10) {
+        this.inProcessFiling = null
```

Comment:
This stops formatting the item differently, right?

Reply:
Yes. If 10 seconds elapses and we haven't reloaded the dashboard, setting this to null passes down to the Todo component and removes the special handling/formatting.